pkg/controller/initializer.go

```diff
@@ -2,11 +2,15 @@ package controller
 
 import (
 	"fmt"
+	"reflect"
 
 	"github.com/RHsyseng/operator-utils/pkg/logs"
 	"github.com/RHsyseng/operator-utils/pkg/utils/kubernetes"
 	"github.com/RHsyseng/operator-utils/pkg/utils/openshift"
+	"github.com/blang/semver"
```

Comment:
Did you check out golang.org/x/mod/semver?

Reply:
tend to use string and with this format (vMajor.Minor : v4.3) for data  transmission?

Reply:
just tested a few things... appears to require a `v` to be prepended to all versions.

Reply:
`v4.3` for example

Reply:
So the built-in semver library requires that, and this one doesn't? I imagine we could work around that by adding the "v" at the beginning, no? I remember reading about plans to adopt semver in OCP, or it may have been somewhere else!

I would do a cost/benefit analysis in this case... the "v" at the beginning being a bit of a hack, but on the other hand, will it be a similar library dependency or a "built-in" feature?

Reply:
i'd agree that if ocp was going to adopt that version of semver... we should too. will try to find out

Reply:
might look something like this ... https://github.com/tchughesiv/kie-cloud-operator/compare/fixVersionCompare...tchughesiv:gSemverNewer

Reply:
Looks good to me! What do you think, @myeung18 ?

Curious, does this avoid vendoring? I'm not sure how the golang/x works

Reply:
still vendors... at least w/ go 1.12 ... maybe that changes in 1.13?

Reply:
For operator-uilts, I will take string as the type to accept version params. Internally, I will verify these input align with mod.semver format, if good, I will trim that 'v' out, and use the OpenShiftVersion to do the rest: do the comparison?  Since in operator-utils, we are still dealing with OpenShiftVersion, does this sound good?

Reply:
any chance we can move OpenShiftVersion type out from internal section so I can use that type in BA operator?

Reply:
@jeremyary will there be any big impact or actually easy to do?

Reply:
@myeung18 not sure I follow. I thought the idea was to use semver to do the comparison. If so, why wouldn't you do that in operator-utils?

Reply:
we can use do that, was just asking, since OpenShiftVersion is now Comparable thought I should make use of it.

Comparsion in operator-uilts will be using mod.semver then.

Reply:
If we think semver makes sense, we should rethink the design of OpenShiftVersion based on that.

Reply:
then the version string return from operator-utils:PlatformInfo, do we keep the existing one or should we make it consistent with this new format?

Reply:
I thought SemVer was what I linked in the previous PR as what I wanted to chase for redesign, but you said it had permissions issues?

Reply:
I haven't looked at it. Please think about it, try to redesign and let's review when ready.

Reply:
> I thought SemVer was what I linked in the previous PR as what I wanted to chase for redesign, but you said it had permissions issues?

https://github.com/RHsyseng/operator-utils/pull/53#discussion_r406310699

Regardless, yes, if this works without elevated perms as I was hoping, it garners a redesign that should do away with some of the complication in operator-utils.

Reply:
@tchughesiv @jeremyary @bmozaffa 
New PR#
RHsyseng/operator-utils#58

Reply:
@jeremyary @myeung18 i think we're getting closer to something usable with that PR, but can someone help me understand the reasoning behind making OpenShiftVersion an `internal` only package? I'd like to import it but am unable.

Reply:
Well, operator-utils has grown a lot since this was implemented last June - this was originally a [refactor of a catch-all 'utils' file](https://github.com/RHsyseng/operator-utils/pull/16/files) that was building up and becoming more deserving of its own package. The original intent when worked out between @bmozaffa and myself was utils.go would put forward an easy-to-read set of endpoints from a single import for the user to consume all of our helper methods without having to filter through implementation (and there were already some projects using method signatures from this utils.go file). There may have been something around the Discoverer interface/testability that led in this direction as well but I can't recall and the convo took place outside of PR's and I can't find it now.

Pretty well everything we've added to the project since hasn't followed this dynamic, likely because our understanding of Go paradigms grew along with the codebase - mine for certain... this feature was where I by and large cut my teeth as a first major go contribution.

Reply:
We probably have better use case to make OpenShifftVersion public and act as carrier.
this how it would look like: https://github.com/RHsyseng/operator-utils/pull/59
platform and openshift pkg are now cross referencing, however. Before it was only one way.

Reply:
Just for the Information, a cut and paste of the pkg from internal to utils pkg will also do the job instead of breaking the current cod. Only if we decided to do so.

Reply:
@tchughesiv @jeremyary @bmozaffa

Reply:
Was Jeremy planning on taking a second look at the whole approach at a high level?